### ISSUE TITLE
Debian Jenkins slave not triggering for runtime repo

### ISSUE THREAD

**Issue by jodh-intel** [kata-containers/ci#128]
Checking a new-ish PR (https://github.com/kata-containers/runtime/pull/1351 from yesterday), no debian CI run occurred fwics. And yet I see debian nodes listed on http://jenkins.katacontainers.io/.

I guess this is because we only have Debian slaves for `master` @chavafg? Is there any reason for this? If we have capacity, please could you add in debian slaves for testing PRs too.

**Comment by chavafg** [comment]
I think Gaby is still working on making debian work smoothly.

@GabyCT can you give us an update?

**Comment by GabyCT** [comment]
@chavafg and @jodh-intel , I tried to enable `Debian 10` on the jenkins CI, however, while launching the VM we have the following error
```
Setting up apt-transport-https (1.8.2) ...
Setting up libyaml-0-2:amd64 (0.2.1-1) ...
Setting up libglib2.0-0:amd64 (2.58.3-2) ...
No schema files found: doing nothing.
Setting up libsasl2-modules:amd64 (2.1.27+dfsg-1) ...
Setting up libnghttp2-14:amd64 (1.36.0-2) ...
Setting up libpackagekit-glib2-18:amd64 (1.1.12-5) ...
Setting up libldap-common (2.4.47+dfsg-3) ...
Setting up libsasl2-modules-db:amd64 (2.1.27+dfsg-1) ...
Setting up libglib2.0-data (2.58.3-2) ...
Setting up librtmp1:amd64 (2.4+20151223.gitfa8646d.1-2) ...
Setting up libsasl2-2:amd64 (2.1.27+dfsg-1) ...
Setting up libgirepository-1.0-1:amd64 (1.58.3-2) ...
Setting up libssh2-1:amd64 (1.8.0-2.1) ...
Setting up libstemmer0d:amd64 (0+svn585-1+b2) ...
Setting up iso-codes (4.2-1) ...
Setting up libpolkit-gobject-1-0:amd64 (0.105-25) ...
Setting up libgstreamer1.0-0:amd64 (1.14.4-1) ...
Setcap worked! gst-ptp-helper is not suid!
Setting up libpolkit-backend-1-0:amd64 (0.105-25) ...
Setting up libglib2.0-bin (2.58.3-2) ...
Setting up libldap-2.4-2:amd64 (2.4.47+dfsg-3) ...
Setting up libcurl3-gnutls:amd64 (7.64.0-4) ...
Setting up libappstream4:amd64 (0.12.5-1) ...
Setting up python3-pycurl (7.43.0.2-0.1) ...
Setting up python3-software-properties (0.96.20.2-2) ...
Setting up gir1.2-glib-2.0:amd64 (1.58.3-2) ...
Setting up libcurl4:amd64 (7.64.0-4) ...
Setting up libpolkit-agent-1-0:amd64 (0.105-25) ...
Setting up curl (7.64.0-4) ...
Setting up policykit-1 (0.105-25) ...
Setting up gir1.2-packagekitglib-1.0 (1.1.12-5) ...
Setting up python3-gi (3.30.4-1) ...
Setting up packagekit (1.1.12-5) ...
Setting up packagekit-tools (1.1.12-5) ...
Setting up software-properties-common (0.96.20.2-2) ...
Processing triggers for libc-bin (2.28-8) ...
Processing triggers for dbus (1.12.12-1) ...
Hit:1 http://debian-archive.trafficmanager.net/debian buster InRelease
Get:2 https://download.docker.com/linux/debian buster InRelease [44.4 kB]
Err:2 https://download.docker.com/linux/debian buster InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 7EA0A9C3F273FCD8
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
Package docker-ce is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source


We trust you have received the usual lecture from the local System
Administrator. It usually boils down to these three things:

    #1) Respect the privacy of others.
    #2) Think before you type.
    #3) With great power comes great responsibility.

+ pkill -f dpkg
+ pkill -f apt-get
+ rm -f /var/lib/dpkg/lock
+ rm -f /var/lib/apt/lists/lock
+ rm -f /var/cache/apt/archives/lock
+ apt clean all

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

+ apt update

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

+ apt install -y git make gcc openjdk-8-jre gnupg2

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

+ apt install -y apt-transport-https ca-certificates curl software-properties-common

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

debconf: unable to initialize frontend: Dialog
debconf: (Dialog frontend will not work on a dumb terminal, an emacs shell buffer, or without a controlling terminal.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.28.1 /usr/local/share/perl/5.28.1 /usr/lib/x86_64-linux-gnu/perl5/5.28 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.28 /usr/share/perl/5.28 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7, <> line 34.)
debconf: falling back to frontend: Teletype
dpkg-preconfigure: unable to re-open stdin: 
+ + curl -fsSL https://download.docker.com/linux/debian/gpg
apt-key add -
E: gnupg, gnupg2 and gnupg1 do not seem to be installed, but one of them is required for this operation
(23) Failed writing body
+ lsb_release -cs
+ add-apt-repository deb [arch=amd64] https://download.docker.com/linux/debian buster stable
+ apt update

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

W: GPG error: https://download.docker.com/linux/debian buster InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 7EA0A9C3F273FCD8
E: The repository 'https://download.docker.com/linux/debian buster InRelease' is not signed.
+ apt install -y docker-ce

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

E: Package 'docker-ce' has no installation candidate
+ gpasswd -a jenkins docker
gpasswd: group 'docker' does not exist in /etc/group
+ chmod g+rw /var/run/docker.sock
chmod: cannot access '/var/run/docker.sock': No such file or directory
+ git config --global user.email katacontainersbot@gmail.com
init.sh: 26: init.sh: git: not found
+ git config --global user.name Kata Containers Bot
init.sh: 27: init.sh: git: not found
+ echo jenkins ALL=(ALL) NOPASSWD: ALL
```
I am installing as a dependencies
```
$ apt install -y git make gcc default-jre openjdk-8-jre apt-transport-https  ca-certificates curl  gnupg2 software-properties-common
```
And I am following the documentation from 
https://docs.docker.com/v17.09/engine/installation/linux/docker-ce/debian/#install-using-the-repository

I tried to perform this locally and is working ok, however, is not working on jenkins

**Comment by GabyCT** [comment]
Closing this issue as is solved
